presto-main/src/main/java/com/facebook/presto/execution/scheduler/NodeScheduler.java

```diff
@@ -159,7 +170,18 @@ public NodeSelector createNodeSelector(ConnectorId connectorId, int maxTasksPerS
                     networkLocationCache);
         }
         else {
-            return new SimpleNodeSelector(nodeManager, nodeSelectionStats, nodeTaskMap, includeCoordinator, nodeMap, minCandidates, maxSplitsPerNode, maxPendingSplitsPerTask, maxTasksPerStage);
+            Collection<InternalNode> nodes = nodeMap.get().getNodesByHostAndPort().values();
+            if (sortedAllNodeList == null || nodes.size() >= sortedAllNodeList.size() && lastNodeListUpdatedTime != null && Duration.nanosSince(lastNodeListUpdatedTime).compareTo(NODE_LIST_REFRESH_INTERVAL) > 0) {
+                // Number of active nodes is greater than or equal to the nodes in allNodeList and its been 10 minutes since last update.
```

Comment:
What if we lose a node for a long time? do we not change the `sortedAllNodeList `?

Reply:
yeah good point. I think it's better to have a cache with TTL, so that we can handle such cases. Will re-write this logic.